### ISSUE TITLE
Discussion of imaging approaches (feather-only vs. joint imaging+feather)

### ISSUE THREAD

**Issue by d-l-walker** [ACES-CMZ/reduction_ACES#384]
#### This issue is to track all relevant discussion related to our approach to obtaining fully combined image cubes. The aim here is to do some tests to get the imaging in the best possible shape, and then determine whether we really need to do the joint imaging, or whether the feather-only approach is good enough.

- We are currently working with mosaics that consist of feathered-only data (feathering 7m+TP, then feathering that product with the 12m).
- We have a subset of the HNCO data for which joint 12m + 7m imaging has been done, which is then feathered with TP data (see #192).
- The joint imaging for the 12m + 7m data currently uses the tclean commands from the 12m-only imaging (except for cyclethreshold, which was sometimes adjusted to avoid divergence).
- It has become apparent that the joint deconvolution approach look worse than the feathering approach. Specifically, there are more significant negative bowls in the jointly imaged data. The initial consensus is that this could be due to either:
     - Insufficient cleaning of the larger scale emission from the 7m data. This is apparent when looking at the residuals (see following comments).
     - An issue with the weighting of the data (7m data over-weighted?)

@d-l-walker will work on some tests to re-do the joint imaging of single channels and sub-cubes with refined masking and deeper cleans to see whether this improves the results.

We should also look into using the TP data as a startmodel for tclean to see whether this improves the results. @d-l-walker will try to look into this too, but feel free to investigate this if you have the time/inclination.

@ashleythomasbarnes has also done some initial tests and comparisons, which will be summarised in a following comment.

**Comment by d-l-walker** [comment]
#### Summary of @ashleythomasbarnes's comparisons:

The Brick (field `ao`, #41) has much more significant negative bowls in the joint 12m/7m + TP feather cube (left) vs. the feathered-only (right).
![Screenshot 2023-09-28 at 09 40 11](https://github.com/ACES-CMZ/reduction_ACES/assets/13015266/9c2d5a8a-f181-45e8-9b24-f866ad57fe33)

For field `i` (#138), the joint 12m/7m + TP feather image (right) appears to have more larger scale emission, but also more significant negatives.
![Screenshot 2023-09-28 at 12 06 23](https://github.com/ACES-CMZ/reduction_ACES/assets/13015266/92004489-f314-40ba-8f49-d4a0e11b0d21)

In the same field, when averaging over an area comparable to the single dish resolution, the fluxes are similar between the two approaches, possibly hinting at the issue being with the joint 12m/7m imaging.
![Screenshot 2023-09-28 at 12 08 23](https://github.com/ACES-CMZ/reduction_ACES/assets/13015266/3182852e-3250-463f-b940-5b149eb9c88f)

Ash testing the following with MOPRA SD data:
1) including the MOPRA data to the cleaned 7m+12m images and it isn’t great, there are still a number of strong negative bowls in the image (hasn’t tested if its better than the TP); 

2) increasing the weighting factor of the single dish in feather (attached show MOPRA+7m+12m; factors of 1, 2, 3, 4, and 5). You can see that upping the factor does solve the bowls, but some other artefacts are present (and the flux gets too high).

![Screenshot 2023-09-28 at 18 18 12](https://github.com/ACES-CMZ/reduction_ACES/assets/13015266/3e3ecc1a-604f-48ba-ab93-5c1782484766)

It was suggested in the WP1 meeting that this should also be tested on the feathered 12m + 7m data, rather that the jointly imaged data.

**Comment by d-l-walker** [comment]
A quick comparison between the jointly imaged 12m/7m data for field `i` (left) vs. the residual (right) is shown below for a single channel. This shows that the bulk of the brighter, more compact emission is being cleaned and captured in the mask, but the larger scale emission is not being cleaned well enough. This could be leading to the poor images that we're currently seeing.

As noted in the parent comment, we will try cleaning deeper and relaxing the masking to include this larger scale emission. This will increase the already lengthly cleaning times, so initial tests will be done on single channels/sub cubes.

![Screenshot 2023-10-04 at 11 23 31](https://github.com/ACES-CMZ/reduction_ACES/assets/13015266/cf2336f6-40ed-42b8-af5a-3dde5008f7b9)

Averaged spectra of the HNCO image and residual for the field shown above.
![Screenshot 2023-10-04 at 11 53 53](https://github.com/ACES-CMZ/reduction_ACES/assets/13015266/c79d8e0e-e90e-4b5d-af87-0ae7e3d2567a)

**Comment by ashleythomasbarnes** [comment]
Here is my cleaning routine I used for Cloud D+E/F 1mm ALMA data. I wrote this a while ago so apologies its a bit of a mess, but all the functionality we discussed should be there, and modules in the directory...   

https://github.com/ashleythomasbarnes/interferometry_analysis/blob/master/casa/cleaning_module.py

**Comment by d-l-walker** [comment]
## Update on joint imaging vs. feathering only

### Summary
* I cleaned a small spectral portion (200 channels) of the HNCO data for field t (#113) at full resolution
* 12m & 7m data are jointly imaged with the TP data as a startmodel, followed by feathering with TP data
* In the below comparisons, this jointly imaged data is on the left, and the feathered only data is on the right 
* The two approaches result in very similar images, but the jointly cleaned image consistently contains more flux, and more extended emission
     * I think this is largely due to better masking and cleaning deeper 
* The negatives that we were previously seeing in the joint imaging are suppressed, and now generally look similar to those in the feather only data
* The joint imaging is very prone to divergence when using the 12m only cleaning parameters from the ALMA PL. Easy solution is to bump up `cyclefactor`, but this takes a long time.
* Next steps:
     * Try on other region(s)
     * Look into speeding things up
     * Discuss how we want to proceed 
* **TL;DR: joint imaging seems to look better, but takes a loooong time**

### Single channel comparisons (note the negatives in jointly imaged data at the lower edge in left panel)
![Screenshot 2024-07-19 at 09 52 09](https://github.com/user-attachments/assets/d8d7e6d8-a5bf-48e7-b50b-b9fb97df45a2)
![Screenshot 2024-07-19 at 09 52 55](https://github.com/user-attachments/assets/2a928a0f-1990-410c-845b-8b3eb84aadd7)
![Screenshot 2024-07-19 at 09 55 15](https://github.com/user-attachments/assets/9626724a-17a7-42a5-8dc2-bb78115ad7a2)

### Some examples of the final image and residual (+ mask contours) for the joint deconvolution. Looks pretty good!
![Screenshot 2024-07-19 at 10 56 33](https://github.com/user-attachments/assets/490e614a-c3a4-4311-8bd5-514bc739ed93)
![Screenshot 2024-07-19 at 10 57 59](https://github.com/user-attachments/assets/4b47a4d8-786d-49a2-b66e-cd2fd117a497)

### Integrated intensity
![Screenshot 2024-07-19 at 11 39 46](https://github.com/user-attachments/assets/3480c639-a953-4207-9965-515d7a0a4fa4)

### Peak intensity
![Screenshot 2024-07-19 at 11 41 08](https://github.com/user-attachments/assets/60681584-e28e-4ffd-85d0-a7527fa24d9d)

### Mean spectra
![Screenshot 2024-07-19 at 13 34 52](https://github.com/user-attachments/assets/ccd22fec-d43a-447f-9a17-67e209c230cf)

### Some quick stats
| cube | min | max | std | sum | mean |
|------|-----|-----|-----|------|------|
| Sgr_A_st_t.TP_7M_12M_feather_all.hnco43.image.statcont.contsub.fits | -0.255 | 0.278 | 0.019 | 715997 | 0.0024 |
| Sgr_A_st_t.hnco43.deep_clean.relaxed_mask.tp_startmodel.image.pbcor.feather.fits | -0.176 | 0.307 | 0.017 | 751702 | 0.0027 |

**Comment by keflavich** [comment]
We probably need to try to quantify 'looks better'.  I think there was a proposed statistic in the Plunkett paper (https://ui.adsabs.harvard.edu/abs/2023PASP..135c4501P/abstract), but I'm not sure that's valuable.

I'd like to know what the effective beam size is.  That's some combination of 'what does the header say' and what is there in reality...

**Comment by keflavich** [comment]
Also it'd be interesting to show the difference image (feathered - jointly cleaned)

**Comment by d-l-walker** [comment]
Sure, I agree. This was just a quick look at some comparisons. I haven't had a chance to look into properly quantifying this yet. 

Here's the difference moments (joint - feathered)
![Screenshot 2024-07-22 at 14 32 11](https://github.com/user-attachments/assets/418eb46f-8677-4f20-a344-3df99bbd582a)

And feathered-joint since that's what you asked for ;)
![Screenshot 2024-07-22 at 14 38 14](https://github.com/user-attachments/assets/ca10499e-26ee-423f-aa71-57cdb2314cd6)

### Restoring beam (joint imaging) = 2.45913" X 2.09486", 74.2558 deg

### Restoring beam (feather only) = 2.42612" X 2.06251", 74.1716 deg
